### ISSUE TITLE
Unable to use different CAs for server / client ingress mTLS

### ISSUE THREAD

**Issue by jwilner** [istio/istio#29856]
**Bug description**

I am trying to build on [the secure ingress docs](https://istio.io/latest/docs/tasks/traffic-management/ingress/secure-ingress/#configure-a-mutual-tls-ingress-gateway) and configure Istio to use a different CA for client verification than that which signed the server cert. Istio does not seem to support this.

Specifically, I have an ingress gateway with a server block like so:
```yaml
...
  - hosts:
    - 'foo.dev'
    port:
      name: https
      number: 443
      protocol: HTTPS
    tls:
      mode: MUTUAL
      credentialName: foo-tls
```

`foo-tls` is created by `cert-manager` and has the fields `tls.crt`, `tls.key`, and `ca.crt`. `ca.crt` _is_ not the `ca.crt` I want to use for clients -- it is the cert of the CA which signed `tls.crt`.

I figured, then, that I could put the alternate CA crt at `foo-tls-cacert`, as the docs indicate. However, even when I do that, istio still prefers the `ca.crt` in `foo-tls`. It is only once I delete the `ca.crt` from `foo-tls` that `foo-tls-cacert` is used and my client certs are accepted.

I would expect the more specific `foo-tls-cacert` to override the `ca.crt` in `foo-tls`. Is there a way to support a separate CA cert for ingress clients without mangling cert-manager's output?

[ ] Docs
[ ] Installation
[x] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure
[ ] Upgrade

**Expected behavior**

Most simply, that istio would prefer  `foo-tls-cacert` over the `ca.crt` field within `foo-tls` for client auth. More generally, that `istio` would support specifying another CA for client auth that played nice with `cert-manager`.

**Steps to reproduce the bug**

- Create the ingress using MUTUAL and `credentialName: foo-tls`, where `foo-tls` has the fields `tls.crt`, `tls.key` and `ca.crt`, where `ca.crt` is the cert of the CA that signed `tls.crt` -- this mirrors `cert-manager`'s output. 
- Put the ca.crt that we want to verify client certs at `foo-tls-cacert` in the field `cacert`
- Try using client certs signed by `foo-tls-cacert` -- confirm failure
- Run `kubectl patch secret -n istio-system foo-tls --type=json -p='[{"op": "remove", "path": "/data/ca.crt"}]'` to delete the obscuring `ca.crt` -- now the curl works.

**Version** (include the output of `istioctl version --remote` and `kubectl version --short` and `helm version --short` if you used Helm)

```shell
$ istioctl version --remote
client version: 1.8.1
control plane version: 1.8.1
data plane version: 1.8.1 (1 proxies)
$ kubectl version --short
Client Version: v1.19.3
Server Version: v1.19.1
```

**How was Istio installed?**

Via istioctl

**Environment where the bug was observed (cloud vendor, OS, etc)**

Local env on `kind`

**Comment by howardjohn** [comment]
I agree this is the desired behavior, and it turns out this actually was the behavior prior to a change we made in Istio 1.8. It also turns out the old behavior was sort of broken as well, if you removed the `-cacerts` Secret it would not fallback to the main Secret - this was already fixed by the change in 1.8

**Comment by jwilner** [comment]
Awesome, thanks @howardjohn.

Qq while I've got you -- do you happen to know if the name from an ingress client cert should be visible to the authorization policies in `source.principals`? There is no mention of it in the docs or anything, but it seems like it should be feasible.
